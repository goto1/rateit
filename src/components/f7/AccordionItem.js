import React from "react";
import PropTypes from "prop-types";

const AccordionItem = ({ className, children }) => {
  const classNames = `accordion-item ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

AccordionItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default AccordionItem;
