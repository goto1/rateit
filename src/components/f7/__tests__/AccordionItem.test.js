import React from "react";
import AccordionItem from "../AccordionItem";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("AccordionItem component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <AccordionItem>
          <span>HelloWorld!</span>
        </AccordionItem>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span", () => {
    const wrapper = shallow(
      <AccordionItem>
        <span>HelloWorld!</span>
      </AccordionItem>
    );
    const actual = wrapper.find("span").at(0).html();
    const expected = "<span>HelloWorld!</span>";
    expect(actual).toBe(expected);
  });

  it("should render with a specified className", () => {
    const wrapper = shallow(
      <AccordionItem className="custom">HelloWorld!</AccordionItem>
    );
    const actual = wrapper.props().className;
    const expected = "accordion-item custom";
    expect(actual).toBe(expected);
  });
});
