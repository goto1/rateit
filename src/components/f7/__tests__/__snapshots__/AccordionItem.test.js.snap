// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem component should render correctly 1`] = `


<div
  className="accordion-item"
>
  <span>
    HelloWorld!
  </span>
</div>
`;
